src/lib/Repository/UiRelationService.php

```diff
@@ -62,17 +62,18 @@ public function loadRelations(VersionInfo $versionInfo)
     /**
      * Loads reverse relations for a piece of content.
      * Retrieves reverse relations and then sets the field definition and the content type name.
+     * Returns false if the user doesn't have permission to load reverse relations.
      *
      * @param ContentInfo $contentInfo
      *
-     * @return UiRelation[]
+     * @return UiRelation[]|bool
      */
     public function loadReverseRelations(ContentInfo $contentInfo)
     {
         try {
             $reverseRelations = $this->contentService->loadReverseRelations($contentInfo);
         } catch (UnauthorizedException $e) {
```

Comment:
The reason it was implemented like this is is to avoid duplicating the code that checks the permissions. So that permissions can remain as part of the kernel and therefore would have one central place where they would need to be edited.